loader/loader.go

```diff
@@ -164,20 +164,24 @@ func IgnoreDotFiles(l *Loader) { l.ignoreDotfiles = true }
 
 func New(runtimePath string, runtimeSubdirectory string, scope stats.Scope, refresher Refresher, opts ...Option) IFace {
 	if runtimePath == "" || runtimeSubdirectory == "" {
-		logger.Warnf("no runtime configuration. using nil loader.")
+		logger.Warn("no runtime configuration. using nil loader.")
 		return NewNil()
 	}
 	watchedPath := refresher.WatchDirectory(runtimePath, runtimeSubdirectory)
 
 	watcher, err := fsnotify.NewWatcher()
 	if err != nil {
-		logger.Fatalf("unable to create runtime watcher %+v", err)
+		// If this fails with EMFILE (0x18) it is likely due to
```

Comment:
If the goal is to make the error more obvious (and make it easier for Future DevEnv to debug same issue), what do you think about showing this directly to the logs, instead of as a comment?  e.g.

```
if err == 0x18 {
    msg = "Unable to create runtime watcher, likely due to inotify_init1() and fs.inotify.max_user_instances %[1]s (%[1]T %#[1]v)\n","
else {
msg =    "unable to create runtime watcher: %[1]s (%[1]T %#[1]v)\n"
}
logger.Panicf(msg, err)
}
```

Reply:
^ formatting got messed up but you get the idea.

Reply:
I'm hesitant to be too specific in the error message since doing so would require additional checks (`runtime.GOOS == "linux"`) and `inotify_init1()` is not the only syscall that `fsnotify.NewWatcher()` makes that will return `EMFILE` on Linux.  Basically, if you see this error you need to do some digging.  Part of that digging should involve tracking down the source of this error and I don't want to lead people in the wrong direction with the error message.

Reply:
@charlievieth  thanks for explaining!